### PR TITLE
Consolidate multi-arch docker builds

### PR DESCRIPTION
**Pull request by ddelange**
- Move docker builds from `release.yaml` to `docker-builds.yaml`
- Delete `republish-images.yaml`:
  - there are no longer single-arch artifacts uploaded.
  - re-trigger by flipping the `this is a prerelease` checkbox on the Github Release

### Example

One commit on top of #7901, diff: https://github.com/ddelange/prefect/compare/docker-builds-release...docker-builds-consolidation

Steps 7-8 ref https://github.com/PrefectHQ/prefect/pull/7740#discussion_r1048803518 for multi arch docker images, closes https://github.com/PrefectHQ/prefect/issues/5388

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by madkinsz** [comment]
Hm, there are some warnings on the latest merge: https://github.com/PrefectHQ/prefect/actions/runs/3950856517

**Comment by madkinsz** [comment]
We've got an example run from a release now: https://github.com/PrefectHQ/prefect/actions/runs/3962751282/jobs/6789813811

**Comment by ddelange** [comment]
nice, the build command in L143 LGTM!

sinve recent versions of the action, there is also `--provenance` being pushed unless disabled, an interesting [feature](https://docs.docker.com/build/attestations/slsa-provenance/) that might break some dev/custom setups ref https://github.com/docker/buildx/issues/1509#issuecomment-1378454396 especially ones that use `docker manifest` commands (my comment further down).

I don't think it causes trouble for `docker pull` though, otherwise you'd probably already have seen a bug report of some sort.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -35,175 +31,26 @@ jobs:
 
       - name: Install python packages
         run: |
-          python -m pip install --upgrade pip
-          pip install --upgrade --upgrade-strategy eager -e .[dev]
+          python -m pip install -U pip setuptools wheel
+          pip install -U -e .[dev]
```
Comment (madkinsz) [inline .github/workflows/release.yaml:35]: Can you revert these changes? I don't think they belong in this pull request.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -226,7 +73,7 @@ jobs:
         run: git checkout -b "docs-for-release-${GITHUB_REF#refs/*/}"
 
       - name: Merge with main
-        run: git merge origin/main --no-ff -Xtheirs -m "Merge branch 'main' into temporary docs branch"
+        run: git merge origin/main --no-ff -m "Merge branch 'main' into temporary docs branch"
```
Comment (madkinsz) [inline .github/workflows/release.yaml:76]: Can you revert this change? This merge strategy is needed and the change doesn't belong here anyway.

Comment (ddelange) [reply .github/workflows/release.yaml:76]: ouch something went wrong with merge conflicts. on it

Comment (madkinsz) [reply .github/workflows/release.yaml]: No problem! Thanks :)

**Review by madkinsz — APPROVED** [review]
Let's give it a go :) Thanks for your hard work on this!
